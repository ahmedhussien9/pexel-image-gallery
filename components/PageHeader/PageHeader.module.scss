.PageHeader {
    padding: 1rem;
    margin: 1rem 0rem;
    background-color: var(--color-background);
    color: var(--color-primary);
    transition: all 0.3s ease-in-out;
    .title {
        color: #b7a210;
    }
}
