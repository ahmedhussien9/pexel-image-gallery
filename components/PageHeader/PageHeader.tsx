import styles from "./PageHeader.module.scss";
type Props = {
    title: string;
    description: string;
}

const PageHeader = ({ title, description }: Props) => (
    <header className={styles.PageHeader} >
        <h1 className={styles.title} aria-label={`page title ${title}`} >{title}</h1>
        <p className={styles.description} aria-label={`page description ${title}`} >{description}</p>
    </header>

)

export default PageHeader
