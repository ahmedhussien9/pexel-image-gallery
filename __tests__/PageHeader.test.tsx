import { render, screen } from '@testing-library/react';
import PageHeader from "../components/PageHeader/PageHeader";

describe('Page Header Component', () => {
    const data = {
        title: "Page title",
        description: "This is a page description"
    };
    it('Should render a snapshot of the page header component', () => {
        const { container } = render(<PageHeader title='Page title' description='This is a page description' />);
        expect(container).toMatchSnapshot();

    });

    it('Should render the title correctly', () => {
        const { container } = render(<PageHeader title='Page title' description='This is a page description' />);
        expect(container.querySelector('.title').innerHTML).toBe("Page title");
    });

    it('Should render the description correctly', () => {
        const { container } = render(<PageHeader title={data.title} description={data.description} />);
        expect(container.querySelector('.description').innerHTML).toBe("This is a page description");
    });
})