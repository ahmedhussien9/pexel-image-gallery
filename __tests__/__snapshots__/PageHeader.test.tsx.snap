// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Header Component Should render a snapshot of the page header component 1`] = `
<div>
  <header
    class="PageHeader"
  >
    <h1
      aria-label="page title Page title"
      class="title"
    >
      Page title
    </h1>
    <p
      aria-label="page description Page title"
      class="description"
    >
      This is a page description
    </p>
  </header>
</div>
`;
